pinot-common/src/main/java/org/apache/pinot/common/helix/ExtraInstanceConfig.java

```diff
@@ -44,4 +45,42 @@ public String getTlsPort() {
   public void setTlsPort(String tlsPort) {
     _proxy.getRecord().setSimpleField(PinotInstanceConfigProperty.PINOT_TLS_PORT.toString(), tlsPort);
   }
+
+  /**
+   * Returns an instance URL from the InstanceConfig. Will set the appropriate protocol and port.
+   */
+  public String getComponentUrl() {
+    String hostName = _proxy.getHostName();
+    // Backward-compatible with legacy hostname of format 'Controller_<hostname>'
```

Comment:
I don't think we allow this legacy hostname format though. Do you see example of this legacy format?

Reply:
I didn't see a specific example of this. I just noticed that other places that were parsing the hostname was doing the same. E.g. [PinotHelixResourceManager](https://github.com/apache/pinot/blob/master/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java#L221)

Reply:
That check is introduced in #5057, and I think we have finished the migration, and this backward compatible handling can be removed

Reply:
Thanks for checking on this Jackie. I went ahead and removed the backward compatible handling.